config/consensus.go

```diff
@@ -463,6 +463,11 @@ type ConsensusParams struct {
 	// the rewardsLevel, but the rewardsLevel has no meaning because the account
 	// has fewer than RewardUnit algos.
 	UnfundedSenders bool
+
+	// TXTailRecentsFix modifies the logic of the txtail's comittedUpTo function to address an off-by-one-error
+	// txtail previously trimmed up to (but not including) round `r + maxlife`. Doing this left 1001 transaction objects after trimming.
+	// the intention is to have only 1000 items after trim, so the logic is modified to trim *including* round `r + maxlife`
+	TXTailRecentsFix bool
```

Comment:
The idea is that if there are different behaviors of TX Tail, transactions would be distributed by some nodes, and discarded by other nodes, since they wouldn't agree on if the lease was still valid on that 1000/1001 threshold. It was originally a discussion with Pavel that prompted this feature flag, but I'll leave it to him for the deeper answer.

Reply:
We need a test to demonstrate this disagreement. I don't think we have one now. At least not a unit test.

Reply:
That could be part of why @algorandskiy had suggested the Protocol flag in the first place -- by enforcing the recents length at the protocol level, there is no concern that peers will disagree, so no test would be needed on that (right?).